improver/fire_weather/drought_code.py

```diff
@@ -0,0 +1,160 @@
+        dc = self._calculate_dc(potential_evapotranspiration)
+
+        return dc
+
+    def _perform_rainfall_adjustment(self):
+        """Updates the previous DC value based on available precipitation
+        accumulation data for the previous 24 hours. This is done element-wise
+        for each grid point.
+
+        From Van Wagner and Pickett (1985), Pages 6-7: Equations 18-21,
+        and Steps 2a-2d corresponding to rainfall adjustment.
+        """
+        # Only adjust if precipitation > 2.8 mm
+        precip_mask = self.precipitation.data > 2.8
+
+        # Step 2a: Calculate effective rainfall via Equation 18
+        effective_rain = 0.83 * self.precipitation.data - 1.27
+
+        # Step 2b: Calculate moisture equivalent of the previous DC via Equation 19
+        # Protect against negative DC values (though unlikely)
+        dc_clipped = np.maximum(self.previous_dc, 0.0)
+        moisture_equivalent = 800.0 * np.exp(-dc_clipped / 400.0)
+
+        # Step 2c: Calculate the moisture equivalent with rainfall via Equation 20
+        # Protect against division by zero in the rain term
+        rain_effect = moisture_equivalent + 3.937 * effective_rain
+        rain_effect = np.clip(rain_effect, a_min=1e-10, a_max=None)
+
+        # Step 2d: Calculate DC after rain via Equation 21
+        dc_after_rain = 400.0 * np.log(800.0 / rain_effect)
+
+        # Step 2d: Apply lower bound of 0
+        dc_after_rain = np.maximum(dc_after_rain, 0.0)
+
+        # Update previous_dc where precipitation > 2.8
+        self.previous_dc = np.where(precip_mask, dc_after_rain, self.previous_dc)
+
+    def _calculate_potential_evapotranspiration(self) -> np.ndarray:
+        """Calculates the potential evapotranspiration adjusted for day length.
+        This represents the moisture loss from deep layers due to evaporation
+        and transpiration.
+
+        From Van Wagner and Pickett (1985), Pages 6-7: Equation 22, Steps 3 & 4.
+
+        Returns:
+            np.ndarray: The potential evapotranspiration value.
+        """
+        # Apply temperature lower bound of -2.8°C
+        temp_adjusted = np.maximum(self.temperature.data, -2.8)
+
+        # Step 3: Get day length factor for current month
+        day_length_factor = self.DC_DAY_LENGTH_FACTORS[self.month]
+
+        # Step 4: Calculate potential evapotranspiration via Equation 22
+        potential_evapotranspiration = 0.36 * (temp_adjusted + 2.8) + day_length_factor
+
+        return potential_evapotranspiration
+
+    def _calculate_dc(self, potential_evapotranspiration: np.ndarray) -> np.ndarray:
+        """Calculates the Drought Code from previous DC and potential evapotranspiration.
+
+        From Van Wagner and Pickett (1985), Page 7: Equation 23.
+
+        Args:
+            potential_evapotranspiration (np.ndarray): The potential evapotranspiration.
+
+        Returns:
+            np.ndarray: The calculated DC value.
```

Comment:
```suggestion
            potential_evapotranspiration: The potential evapotranspiration.

        Returns:
            The calculated DC value.
```

Type not required in docstring as it is expressed in type hint.